// Reducers/cart.js
import {ADD_CART, DELETE_CART} from 'constans.js'

const initialState = {
	sum: 0,
	products: {
		id1: {
			id1001: 4,
			id1002: 4
		}
	}
}

export default (state = initialState, action) => {
	const {type, payload} = action
	switch( type ){
		case ADD_CART:
			const {price, productId, offerId} = payload

			function bodyProductId(){
				return ({
					...state.products[productId],
					[offerId]: bodyOfferId()
				})
			}
			function bodyOfferId(){
				if( !state.products[productId] || !state.products[productId][offerId] )
					return 1
				return state.products[productId][offerId] + 1
			}
			return {
				sum: state.sum + (+payload.price),
				products: {
					...state.products, 
					[productId]: bodyProductId()
				}
			}
		case DELETE_CART:

			const deleteOffer = () => {
				var obj = Object.keys(state.products[payload.id]).reduce( (acc, offer) => {
					if( offer != payload.offerId ){
						console.log(offer, payload.offerId)
						obj = {
							...obj,
							[offer]: state.products[payload.id][offer]
						}
					}
					console.log(obj)
					return obj
				}, {})
				return obj
			}

			return {
				...state,
				products: {
					...state.products,
					[payload.id]: deleteOffer()
				}
			}
		default:
			return state
	}
}

//[payload.productId]: state.ids[payload.id] ? state.ids[payload.id] + 1 : 1
//
