// Components/Cart/index.js
import React, {Component} from 'react'
import {connect} from 'react-redux'
import {Link} from 'react-router-dom'
import RaisedButton from 'material-ui/RaisedButton'
import FontIcon from 'material-ui/FontIcon'
import CircularProgress from 'material-ui/CircularProgress'

import './style.css'
import {loadProduct} from 'Actions'

class Cart extends Component{
	render(){
		
const style = {
  margin: 12,
};
		const {cart, product, loadProduct} = this.props
		function getBody(){
				if(!Object.keys(cart.products).length) return <b>The basket is empty.<br /> <Link to="/">Catalog</Link></b>
				return Object.keys(cart.products).map( (id, i) => {
					console.log(id)

					if( !product[id] ){
						loadProduct(id)
						return <CircularProgress size={60} thickness={7} />
					}

					const {prop} = product[id]

					return(
						<div key={id} className="cart-path">
							<div className="cart-path__img">
								<img src={prop.image} alt={prop.name} />
							</div>
							<div className="cart-path__name">
								<h3><Link to={'/phones/'+prop.id}>{prop.name}</Link></h3>
							</div>
							<div className="cart-path__count">
							</div>
							<div className="cart-path__delete">
								<button>X</button>
								<RaisedButton
									href="https://github.com/callemall/material-ui"
									target="_blank"
									secondary={true}
									icon={<FontIcon className="muidocs-icon-custom-github" />}
									style={style}
								/>
							</div>
						</div>
					)
				})
		}
		return(
			<div className="cart">
				<div className="container">
					<h1>Cart</h1>
					<div className="cart-table">
						{getBody()}
					</div>
				</div>
			</div>
		)
	}
}

const mapStateToProps = state => ({ 
	cart: state.cart,
	product: state.product
})

const mapDispatchToProps = dispatch => ({
	loadProduct: (id) => dispatch(loadProduct(id))
})


export default connect(mapStateToProps, mapDispatchToProps)(Cart)
