// Components/Cart/index.js
import React, {Component} from 'react'
import {connect} from 'react-redux'
import {Link} from 'react-router-dom'
import RaisedButton from 'material-ui/RaisedButton'
import FontIcon from 'material-ui/FontIcon'

import './style.css'

class Cart extends Component{
	render(){
		
const style = {
  margin: 12,
};
		const {cart, phones} = this.props
		function getBody(){
				if(!Object.keys(cart.ids).length) return <b>The basket is empty.<br /> <Link to="/">Catalog</Link></b>
				return Object.keys(cart.ids).map( id => {
					console.log( Object.keys(cart.ids) )
					console.log(id)
					return(
						<div key={id} className="cart-path">
							<div className="cart-path__img">
								<img src={phones[id].image} alt={phones[id].name} />
							</div>
							<div className="cart-path__name">
								<h3><Link to={'/phones/'+phones[id].id}>{phones[id].name}</Link></h3>
							</div>
							<div className="cart-path__count">
								Count: {cart.ids[id]}
							</div>
							<div className="cart-path__delete">
								<button>X</button>
    <RaisedButton
      href="https://github.com/callemall/material-ui"
      target="_blank"
      secondary={true}
      icon={<FontIcon className="muidocs-icon-custom-github" />}
      style={style}
    />
							</div>
						</div>
					)
				})
		}
		return(
			<div className="cart">
				<div className="container">
					<h1>Cart</h1>
					<div className="cart-table">
						{getBody()}
					</div>
				</div>
			</div>
		)
	}
}

const mapStateToProps = state => ({ 
	cart: state.cart,
	phones: state.loadProducts.phones
})

export default connect(mapStateToProps)(Cart)
