// Components/ProductListItem/index.js
import React, {Component} from 'react'
import PropTypes from 'prop-types'
import RaisedButton from 'material-ui/RaisedButton'
import {connect} from 'react-redux'

import './style.css'

class ProductListItem extends Component{
	render(){

		const {id, name, link, srcImg, price} = this.props
		const style = {
			margin: 12,
		};
		return(
			<div className="product-list-item">
				<a href={`#$(id)`} className="product-list-item__img">
					<img src={srcImg} />
				</a>
				<a href={link} className="product-list-item__name">{name}</a>
				<div className="product-list-item__price">$ {price}</div>
				<RaisedButton className="product-list-item__btn" label="Add cart" primary={true} onClick={this.props.funcAddCart.bind(null, price, id)} />
			</div>
		)
	}
}

ProductListItem.propTypes = {
	product: PropTypes.shape({
		id: PropTypes.number.isRequired,
		price: PropTypes.number.isRequired,
		name: PropTypes.string.isRequired,
		link: PropTypes.string.isRequired,
		srcImg: PropTypes.string.isRequired,
		funcAddCart: PropTypes.func.isRequired
	})
}

export default connect(null)(ProductListItem)
