// Components/ProductListItem/index.js
import React, {Component} from 'react'
import PropTypes from 'prop-types'
import RaisedButton from 'material-ui/RaisedButton'
import {connect} from 'react-redux'
import {Link} from 'react-router-dom'

import './style.css'

class ProductListItem extends Component{
	render(){

		const {id, name, srcImg, price} = this.props

		return(
			<div className="product-list-item">
				<Link to={"/phones/"+id} className="product-list-item__img">
					<img src={srcImg} alt={name} />
				</Link>
				<Link to={"/phones/"+id} className="product-list-item__name">{name}</Link>
				{/*<div className="product-list-item__price">$ {price}</div>*/}
				<Link to={"/phones/"+id} >
					<RaisedButton className="product-list-item__btn" label="More info" primary={true} style={{width: '100%'}} />
				</Link>
				{/*<RaisedButton className="product-list-item__btn" label="Add cart" primary={true} onClick={this.props.funcAddCart.bind(null, price, id)} />*/}
			</div>
		)
	}
}

ProductListItem.propTypes = {
	product: PropTypes.shape({
		id: PropTypes.number.isRequired,
		price: PropTypes.number.isRequired,
		name: PropTypes.string.isRequired,
		link: PropTypes.string.isRequired,
		srcImg: PropTypes.string.isRequired,
		funcAddCart: PropTypes.func.isRequired
	})
}

export default connect(null)(ProductListItem)
