// Components/Catalog/index.js
import React, {Component} from 'react'
import {connect} from 'react-redux'
import {loadCatalog, addCart} from 'Actions'
import ProductListItem from 'Components/ProductListItem'
import CircularProgress from 'material-ui/CircularProgress'

class Catalog extends Component{
	componentDidMount(){
		if( !this.props.loaded && !this.props.loading )
			this.props.loadCatalog()
	}
	render(){

		if( this.props.error )
			return <h1>Server error</h1>

		const {loading, loaded, phones, addCart} = this.props

		function getBody(){
			if( loading && !loaded ) return <CircularProgress size={60} thickness={7} />
			return(
				<div className="row">
					{phones.map((phone) => {
						return(
							<div className="col-md-4" key={phone.id}>
								<ProductListItem 
									id={phone.id}
									name={phone.name}
									link={phone.id}
									price={phone.price}
									srcImg={phone.image}
									funcAddCart={ addCart }
								/>
							</div>
						)
					})}
				</div>
			)
		}
		return(
			<div>
				{getBody()}
			</div>
		)
	}
}


const mapDispatchToProps = (dispatch) => {
	return({
		loadCatalog: () => dispatch( loadCatalog() ),
		addCart: (price, id) => dispatch( addCart(price, id) )
	})
}
const mapStateToProps = (state) => {
	return{
		loading: state.catalog.loading,
		loaded: state.catalog.loaded,
		error: state.catalog.error,
		phones: Object.values(state.catalog.phones)
	}
}

export default connect(mapStateToProps, mapDispatchToProps)(Catalog)
