// Components/Catalog/index.js
import React, {Component} from 'react'
import {connect} from 'react-redux'
import {loadProducts} from 'Actions'
import ProductListItem from 'Components/ProductListItem'

class Catalog extends Component{
	componentDidMount(){
		if( !this.props.loaded || this.props.loading )
			this.props.loadProducts()
	}
	render(){
		const {loading, loaded, phones} = this.props
		function getBody(){
			if( loading && !loaded ) return <p>Loading</p>
			return(
				<div className="row">
					{phones.map((phone) => {
						return(
							<div className="col-md-4">
								<ProductListItem 
									key={phone.id}
									id={phone.id}
									name={phone.name}
									link={phone.id}
									srcImg={phone.image}
								/>
							</div>
						)
					})}
		
				</div>
			)
		}
		return(
			<div>
				{getBody()}
			</div>
		)
	}
}


const mapDispatchToProps = (dispatch) => {
	return({
		loadProducts: () => dispatch( loadProducts() )
	})
}
const mapStateToProps = (state) => {
	return{
		loading: state.loadProducts.loading,
		loaded: state.loadProducts.loaded,
		phones: state.loadProducts.phones
	}
}

export default connect(mapStateToProps, mapDispatchToProps)(Catalog)
