// Components/DetailedPhone/index.js
import React, {Component} from 'react'
import {connect} from 'react-redux'
import CircularProgress from 'material-ui/CircularProgress'
import {loadProducts} from 'Actions'
import RaisedButton from 'material-ui/RaisedButton'
import {Link} from 'react-router-dom'

import './style.css'
import {addCart} from 'Actions'

class DetailedPhone extends Component{
	constructor(props){
		super(props);
		this.state = {
			addProduct: false
		}
	}

	componentDidMount(){
		if( !this.props.loaded && !this.props.loading )
			this.props.loadProducts()
	}
	addCart = () => {
		this.setState(prevState => {
			return {addProduct: true}
		})
		this.props.addCart( this.props.phones[this.props.match.params.id].price ,this.props.match.params.id)
	}
	render(){
		const id = this.props.match.params.id
		const {loading, loaded, phones} = this.props
		const phone = phones[id]
		console.log(phone)
		function getBody(){
			if( loading || !loaded ) 
				return <CircularProgress size={60} thickness={7} />
			const {characteristics, description} = phone
			return(
				<div className="b-detailed-phone">
					<div className="row">
						<div className="col-md-6">
							<div className="b-detailed-phone__img txt-c">
								<img src={phone.image} />
							</div>
						</div>
						<div className="col-md-6">
							<h1>{phone.name}</h1>
							<div className="b-detailed-phone__path">
								<ul className="b-detailed-phone__characteristics">
									{characteristics.map( (characteristic, i) => {
										return <li key={i}><b>{characteristic.name}</b>: {characteristic.value}</li>
										})
									}
								</ul>
							</div>
							<div className="b-detailed-phone__path">
								{description}
							</div>
						</div>
					</div>
				</div>
			)
		}
		return(
			<div>
				<div className="container">
					<div className="row">
						<div className="col-md-9">
							{getBody()}
						</div>
						<div className="col-md-3">
							<RaisedButton 
								label={this.state.addProduct ? 'Added' : 'Add cart'} 
								style={{width: '100%', marginBottom: '15px'}} 
								primary={ this.state.addProduct ? false : true} 
								secondary={ this.state.addProduct ? true : false} 
								onClick={this.addCart}  
							/>
							<Link to="/cart">
								<RaisedButton 
									label='Go to cart' 
									style={{width: '100%', marginBottom: '15px'}} 
									secondary={true} 
									onClick={this.addCart}  
								/>
							</Link>
						</div>
					</div>
				</div>
			</div>
		)
	}
}

const mapStateToProps = (state) => {
	return({
		loading: state.loadProducts.loading,
		loaded: state.loadProducts.loaded,
		phones: state.loadProducts.phones
	})
}

const mapDispatchToProps = (dispatch) => {
	return({
		loadProducts: () => dispatch(loadProducts()),
		addCart: (id) => dispatch(addCart(id))
	})
}

export default connect(mapStateToProps, mapDispatchToProps)(DetailedPhone)
